api/sections/add_section_ids.py

```diff
@@ -37,21 +35,54 @@ def main():
     q = qs.API(schoolcode)
     row_num = 1
 
-    for csv_section_info in qs.bar(csv_sections):
-        section_name = csv_section_info['Section Name']
-        student = csv_section_info['Student ID']
-        row_num = row_num + 1
-
-        section = q.match_section(
-            identifier=section_name,
-            student_id=student,
-            match_name=True)
-        section_id = section['id']
-        csv_section_info['Section ID'] = section_id
-        qs.logger.info({"row_num": row_num,
-                        "section_name": section_name,
-                        "student": student,
-                        "section_id": section_id}, cc_print=True)
+    if 'Student ID' in csv_sections.cols:
+        qs.logger.info('Retrieving sections from csv & matching with db...',
+                       cc_print=True)
+        for csv_section_info in qs.bar(csv_sections):
+            section_name = csv_section_info['Section Name']
+            student = csv_section_info['Student ID']
+            row_num = row_num + 1
+
+            section = q.match_section(
+                identifier=section_name,
+                student_id=student,
+                match_name=True)
+            section_id = section['id']
+            csv_section_info['Section ID'] = section_id
+
+    elif 'Section Code' in csv_sections.cols:
+            sections = {}
+            sections_with_id = {}
+            row_num = 1
+
+            qs.logger.info('Retrieving section code from csv...',
+                           cc_print=True)
+            for csv_section_info in csv_sections:
+                section_name = csv_section_info[u'Section Name']
+                section_code = csv_section_info[u'Section Code']
```

Comment:
stick to either `''` or `u''`, but best not to switch. Personally, unless I'm specifically doing something unicode-specific, I just do `''`.